fintual-oss/paraffin-back
# feat: add google sign in with oauth2

# Base PR
- PR based on main

# Context
- With this PR, it is possible to login with google with omniauth gem

# Changelog

- Add new gem: omniaut for manage multi-provider authentication
- Add new gem: omniauth-google-oauth2 for authenticate with Google via OAuth2 in OmniAuth
- Add new gem: omniauth-rails_csrf_protection to handle csrf protection
- Create a new view for sign in
- Create a new view for sign up
- Create a new controller for omniauth callbacks
- Create a new controller for registrations
- Create a new controller for sessions
- Create a new method for user model
- Create migrations for add provider and oauth_id to user model


# QA
- Check if your email is on the list of Google Cloud testers
- Create .env file with the credentials
- Run: `docker compose build`
- Run: `docker compose run web rails db:reset`
- Run: `docker compose run web rails db:migrate`
- [FRONT] On the branch `feature/googleLogin` go to `http://localhost:3001/users/sign_in`
